### PR TITLE
Proposal ref(*): use alpine as replacement of ubuntu-debootstrap as base image

### LINKED ISSUE #3514: InsecurePlatformWarning during builds
Recently urllib3 began [emitting these warnings](https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning) during our normal builds whenever python touches SSL. Looks like we can fix it by updating to Python 2.7.9 (or by explicitly silencing it). An example:
```
Processing triggers for libc-bin (2.19-0ubuntu6.6) ...
/tmp/tmp8K2oTh/pip.zip/pip/_vendor/requests/packages/urllib3/util/ssl_.py:79: InsecurePlatformWarning: 
A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and 
may cause certain SSL connections to fail. For more information, see 
https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
```

### PR DESCRIPTION
**Pull request by aledbf**
replaces #3175 

*//edit carmstrong*
closes #3514

### CODE REVIEW

File: builder/image/Dockerfile [hunk 1]
```diff
@@ -13,28 +11,32 @@ RUN curl -sSL -o /usr/local/bin/etcdctl https://s3-us-west-2.amazonaws.com/opdem
...
+  chmod +x /usr/bin/docker
 
 # configure ssh server
-RUN rm /etc/ssh/ssh_host_*
-RUN dpkg-reconfigure openssh-server
-RUN mkdir -p /var/run/sshd
-
-# configure locale
-RUN echo LANG="en_US.UTF-8" > /etc/default/locale && dpkg-reconfigure locales
+RUN mkdir -p /var/run/sshd && rm -rf /etc/ssh/ssh_host*
```
Comment (carmstrong) [inline builder/image/Dockerfile]: Should we move this to bin/boot in a separate commit so we can also close #3132?

File: builder/image/etc/ssh/sshd_config [hunk 2]
```diff
@@ -23,8 +23,6 @@ X11Forwarding no
 X11DisplayOffset 10
 PrintMotd no
 PrintLastLog yes
-TCPKeepAlive yes
```
Comment (carmstrong) [inline builder/image/etc/ssh/sshd_config]: Why are we removing these?

File: database/bin/boot [hunk 3]
```diff
@@ -71,32 +71,32 @@ INIT_ID=$(etcdctl get $ETCD_PATH/initId 2> /dev/null || echo none)
 echo "database: expecting initialization id: $INIT_ID"
 
 initial_backup=0
-if [[ "$(cat /var/lib/postgresql/9.3/main/initialized 2> /dev/null)" != "$INIT_ID" ]]; then
+if [[ "$(cat /var/lib/postgresql/main/initialized 2> /dev/null)" != "$INIT_ID" ]]; then
```
Comment (carmstrong) [inline database/bin/boot]: I'm worried about the upgrade path here - for users running the old version of Deis before upgrading database here, their `initialized` file will be in the old location.

File: tests/fixtures/test-etcd/Dockerfile [hunk 4]
```diff
@@ -12,9 +10,8 @@ RUN curl -sSL -o /usr/local/bin/etcdctl https://s3-us-west-2.amazonaws.com/opdem
 # build etcd and then clean up
 RUN buildDeps='curl git-core'; \
     set -x; \
-    export DEBIAN_FRONTEND=noninteractive; \
     apt-get update && apt-get install -y $buildDeps --no-install-recommends \
-    && curl -sSL https://storage.googleapis.com/golang/go1.4.1.linux-amd64.tar.gz | tar -C /usr/local -xz \
+    && curl -sSL https://storage.googleapis.com/golang/go1.4.2.linux-amd64.tar.gz | tar -C /usr/local -xz \
```
Comment (carmstrong) [inline tests/fixtures/test-etcd/Dockerfile]: This should probably be a separate commit.

**Comment by carmstrong** [comment]
Good work, @aledbf! :+1:

File: tests/fixtures/test-etcd/Dockerfile (hunk 4 above)
Comment (aledbf) [inline tests/fixtures/test-etcd/Dockerfile]: done

**Comment by mboersma** [comment]
This also closes #3514.

**Comment by carmstrong** [comment]
Code LGTM. So store remains using `ubuntu-debootstrap` for the foreseeable future? Is there any significant downside to having two different base images to the time being?

**Comment by aledbf** [comment]
> So store remains using ubuntu-debootstrap for the foreseeable future?

I think so. The new gliderlabs alpine image now contains a glibc wrapper for a lot of the missing calls in musl but given the role and importance of Ceph in the cluster I prefer to keep using `ubuntu-debootstrap`

> Is there any significant downside to having two different base images to the time being?

I don't think so because the alpine base image is just 5MB.

**Comment by carmstrong** [comment]
:+1: Code LGTM.

**Comment by mboersma** [comment]
This looks good, but I'm doing more manual testing to reassure myself.

**Comment by mboersma** [comment]
Code LGTM. Tested manually, especially `deis pull`. Nice work @aledbf!
```console
REPOSITORY                   TAG         IMAGE ID            CREATED              VIRTUAL SIZE
deis/logspout                alpine      ddf35222a4b9        About a minute ago   10.48 MB
deis/logspout                ubuntu      bd5b6ec54259        7 minutes ago        7.882 MB
deisci/builder               alpine      d051447050fc        About a minute ago   117.7 MB
deisci/builder               ubuntu      4046f9cebd6b        13 minutes ago       362.4 MB
deisci/cache                 alpine      921dfc41710f        About a minute ago   14.86 MB
deisci/cache                 ubuntu      5cce8c96c565        12 minutes ago       95.87 MB
deisci/controller            alpine      b2f4aa270860        54 minutes ago       132 MB
deisci/controller            ubuntu      c55389eac88a        10 minutes ago       297.9 MB
deisci/database              alpine      359c2d1fa9c0        51 minutes ago       181.1 MB
deisci/database              ubuntu      b7ec89224f8b        7 minutes ago        388.9 MB
deisci/logger                alpine      e22d9c0c6f00        About a minute ago   9.887 MB
deisci/logger                ubuntu      2b81c94a62e6        7 minutes ago        7.288 MB
deisci/publisher             alpine      df4fce44f8cb        About a minute ago   10.12 MB
deisci/publisher             ubuntu      c76ab340ef3e        7 minutes ago        7.525 MB
deisci/registry              alpine      d5d81d7cc201        49 minutes ago       155.3 MB
deisci/registry              ubuntu      7fd5602327f3        4 minutes ago        217.8 MB
deisci/router                alpine      f7f11521c2ed        59 seconds ago       31.72 MB
deisci/router                ubuntu      36af05eab446        About a minute ago   152.1 MB
deisci/store-admin           alpine      56b2de9f0b36        4 hours ago          365.1 MB
deisci/store-admin           ubuntu      56b2de9f0b36        4 hours ago          365.1 MB
deisci/store-base            alpine      bd054f97ed0f        4 hours ago          365.1 MB
deisci/store-base            ubuntu      bd054f97ed0f        4 hours ago          365.1 MB
deisci/store-daemon          alpine      b236edae91d1        4 hours ago          365.1 MB
deisci/store-daemon          ubuntu      b236edae91d1        4 hours ago          365.1 MB
deisci/store-gateway         alpine      4a3005161651        4 hours ago          384.3 MB
deisci/store-gateway         ubuntu      4a3005161651        4 hours ago          384.3 MB
deisci/store-metadata        alpine      ab373916f187        4 hours ago          365.1 MB
deisci/store-metadata        ubuntu      ab373916f187        4 hours ago          365.1 MB
deisci/store-monitor         alpine      5f4bd8c1000d        4 hours ago          365.1 MB
deisci/store-monitor         ubuntu      5f4bd8c1000d        4 hours ago          365.1 MB
deisci/swarm                 alpine      6756335d5065        43 seconds ago       16.75 MB
deisci/swarm                 ubuntu      c42039877ced        About a minute ago   14.15 MB
```
